src/native/crashtracker.rs

```diff
@@ -0,0 +1,283 @@
+
+static CRASHTRACKER_STATUS: AtomicU8 = AtomicU8::new(CrashtrackerStatus::NotInitialized as u8);
+static INIT: Once = Once::new();
+
+#[pyfunction(name = "crashtracker_init")]
+pub fn crashtracker_init<'py>(
+    config: &Bound<'py, CrashtrackerConfigurationPy>,
+    receiver_config: &Bound<'py, CrashtrackerReceiverConfigPy>,
+    metadata: &Bound<'py, MetadataPy>,
+) -> PyResult<()> {
+    INIT.call_once(|| {
+        let (inner_config, inner_receiver_config, inner_metadata) = (
+            config.borrow_mut().deref_mut().take_inner(),
+            receiver_config.borrow_mut().deref_mut().take_inner(),
+            metadata.borrow_mut().deref_mut().take_inner(),
+        );
+
+        if let (Some(config), Some(receiver_config), Some(metadata)) =
+            (inner_config, inner_receiver_config, inner_metadata)
+        {
+            match datadog_crashtracker::init(*config, *receiver_config, *metadata) {
+                Ok(_) => CRASHTRACKER_STATUS
+                    .store(CrashtrackerStatus::Initialized as u8, Ordering::SeqCst),
+                Err(e) => {
+                    eprintln!("Failed to initialize crashtracker: {}", e);
+                    CRASHTRACKER_STATUS.store(
+                        CrashtrackerStatus::FailedToInitialize as u8,
+                        Ordering::SeqCst,
+                    );
+                }
+            }
+        } else {
+            eprintln!("Failed to initialize crashtracker: malformed configuration");
+            CRASHTRACKER_STATUS.store(
+                CrashtrackerStatus::FailedToInitialize as u8,
+                Ordering::SeqCst,
+            );
+        }
+    });
+    Ok(())
+}
+
+#[pyfunction(name = "crashtracker_on_fork")]
+pub fn crashtracker_on_fork<'py>(
+    config: &Bound<'py, CrashtrackerConfigurationPy>,
+    receiver_config: &Bound<'py, CrashtrackerReceiverConfigPy>,
+    metadata: &Bound<'py, MetadataPy>,
+) -> PyResult<()> {
+    let inner_config: Box<CrashtrackerConfiguration> = (*config.borrow_mut().deref_mut())
+        .take_inner()
+        .ok_or_else(|| {
+            PyErr::new::<pyo3::exceptions::PyRuntimeError, _>("CrashtrackerConfiguration is None")
+        })?;
+    let inner_receiver_config: Box<CrashtrackerReceiverConfig> = (*receiver_config
+        .borrow_mut()
+        .deref_mut())
+    .take_inner()
+    .ok_or_else(|| {
+        PyErr::new::<pyo3::exceptions::PyRuntimeError, _>("CrashtrackerReceiverConfig is None")
+    })?;
+    let inner_metadata: Box<Metadata> = (*metadata.borrow_mut().deref_mut())
+        .take_inner()
+        .ok_or_else(|| PyErr::new::<pyo3::exceptions::PyRuntimeError, _>("Metadata is None"))?;
+
+    // Note to self: is it possible to call crashtracker_on_fork before crashtracker_init?
+    // dd-trace-py seems to start crashtracker early on.
+    datadog_crashtracker::on_fork(*inner_config, *inner_receiver_config, *inner_metadata)
+        .map_err(|e| PyErr::new::<pyo3::exceptions::PyRuntimeError, _>(e.to_string()))
```

Comment:
Is there a way to shorten this whole `.map_err(...)`. trait on the result type or use a macro